RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT_vect.cc

```diff
@@ -0,0 +1,1032 @@
+#include "RecoVertex/PrimaryVertexProducer/interface/DAClusterizerInZT_vect.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "DataFormats/GeometryCommonDetAlgo/interface/Measurement1D.h"
+#include "RecoVertex/VertexPrimitives/interface/VertexException.h"
+
+#include <cmath>
+#include <cassert>
+#include <limits>
+#include <iomanip>
+#include "FWCore/Utilities/interface/isFinite.h"
+#include "vdt/vdtMath.h"
+
+using namespace std;
+
+namespace {
+  constexpr double epsilon = 1.0e-3;
+  constexpr double vertexSizeTime = 0.008;
+  constexpr double dtCutOff = 4.0;
+}
+
+DAClusterizerInZT_vect::DAClusterizerInZT_vect(const edm::ParameterSet& conf) {
+
+  // hardcoded parameters
+  maxIterations_ = 100;
+  mintrkweight_ = 0.5; // conf.getParameter<double>("mintrkweight");
+
+
+  // configurable debug outptut debug output
+  verbose_ = conf.getUntrackedParameter<bool> ("verbose", false);
+  zdumpcenter_ = conf.getUntrackedParameter<double> ("zdumpcenter", 0.);
+  zdumpwidth_ = conf.getUntrackedParameter<double> ("zdumpwidth", 20.);
+  
+  // configurable parameters
+  double Tmin = conf.getParameter<double> ("Tmin")*std::sqrt(2.0);
+  double Tpurge = conf.getParameter<double> ("Tpurge")*std::sqrt(2.0);
+  double Tstop = conf.getParameter<double> ("Tstop")*std::sqrt(2.0);
+  vertexSize_ = conf.getParameter<double> ("vertexSize");
+  coolingFactor_ = conf.getParameter<double> ("coolingFactor");
+  useTc_=true;
+  if(coolingFactor_<0){
+    coolingFactor_=-coolingFactor_; 
+    useTc_=false;
+  }
+  coolingFactor_ = std::sqrt(coolingFactor_);
```

Comment:
looks like another source of confusion for the future where the meaning of the input parameter is changed after reading it.

Reply:
is there an answer to this?